src/oph/ehoks/opiskelijapalaute.clj

```diff
@@ -52,23 +58,29 @@
         :else true)))
   ([kysely current-hoks updated-hoks] ; on HOKS update
     {:pre [(#{:aloituskysely :paattokysely} kysely)]}
-    ; On HOKS update, when not sending a kysely, log prints are more or less
-    ; redundant because the relevant parameters are unlikely to change."
-    (and (:osaamisen-hankkimisen-tarve updated-hoks)
-         (not (c/tuva-related-hoks? updated-hoks))
-         (if (= kysely :aloituskysely)
-           ; Send aloituskysely if `osaamisen-hankkimisen-tarve` was
-           ; `false` or `nil` previously, ...
-           (or (not (:osaamisen-hankkimisen-tarve current-hoks))
-               ; ... or if email address is added.
-               (and (some? (:sahkoposti updated-hoks))
-                    (nil? (:sahkoposti current-hoks)))
-               ; ... or if phone number is added.
-               (and (some? (:puhelinnumero updated-hoks))
-                    (nil? (:puhelinnumero current-hoks))))
-           ; Send päättökysely if `osaamisen-hankkimisen-pvm` has been added.
-           (and (nil? (:osaamisen-saavuttamisen-pvm current-hoks))
-                (some? (:osaamisen-saavuttamisen-pvm updated-hoks)))))))
+    (let [msg (format "Sending %s for HOKS `%s`. "
+                      (name kysely) (:id updated-hoks))]
+      (and (:osaamisen-hankkimisen-tarve updated-hoks)
+           (not (c/tuva-related-hoks? updated-hoks))
+           (if (= kysely :aloituskysely)
```

Comment:
(idea) Vois itse asiassa olla luettavuuden kannalta parempi, että tätä (if):a ei olisi ollenkaan vaan sen haarat yhdistettäis samaan (cond):iin ja jos ne liittyy ainoastaan tiettyyn kyselytyyppiin, niin sitten olis siellä lisäehtona `(and (= kysely :aloituskysely)` tjsp.